### PR TITLE
Added the missing cachePartitionLostListenerConfigs to CacheConfig serialization and client protocol

### PR DESCRIPTION
**Pull request by ihsandemir**
Added the cachePartitionLostListenerConfigs to CacheConfig serialization and the client protocol encoding.

### CODE REVIEW

File: hazelcast/src/main/java/com/hazelcast/client/impl/protocol/codec/holder/CacheConfigHolder.java [hunk 1]
```diff
@@ -221,6 +230,13 @@ public boolean isDisablePerEntryInvalidationEvents() {
         config.setMergePolicyConfig(mergePolicyConfig);
         config.setDisablePerEntryInvalidationEvents(disablePerEntryInvalidationEvents);
 
+        if (cachePartitionLostListenerConfigs != null) {
+            List<CachePartitionLostListenerConfig> partitionLostListenerConfigs = new ArrayList<>(
+                    cachePartitionLostListenerConfigs.size());
+            cachePartitionLostListenerConfigs.forEach(listenerConfigHolder -> partitionLostListenerConfigs
+                    .add(listenerConfigHolder.asListenerConfig(serializationService)));
```
Comment (blazember) [inline hazelcast/src/main/java/com/hazelcast/client/impl/protocol/codec/holder/CacheConfigHolder.java]: It seems we create and fill a temporary `ArrayList` here that we drop after filling without using it. I guess we miss a `config.setPartitionLostListenerConfigs()` call here.

Comment (ihsandemir) [reply hazelcast/src/main/java/com/hazelcast/client/impl/protocol/codec/holder/CacheConfigHolder.java]: Good catch. Fixed it.

**Comment by ihsandemir** [comment]
run-lab-run

**Comment by ihsandemir** [comment]
run-lab-run

**Comment by ihsandemir** [comment]
run-lab-run

**Pull request merged**